From daae7d9e0da9c6ba52dcfac8f4901a3f1330582b Mon Sep 17 00:00:00 2001
From: Ashok Kumar <ashok2.kumar@amd.com>
Date: Tue, 24 May 2022 07:09:40 -0500
Subject: [PATCH 1/1] webUI changes to display CPU information

webUI changes to display CPU information Model ID,
Family ID and Stepping ID for each socket.

Signed-off-by: Ashok Kumar <ashok2.kumar@amd.com>

%% original patch: 0011-webUI-changes-o-display-CPU.patch
---
 src/locales/en-US.json                        |  5 ++
 .../ResourceManagement/PowerControlStore.js   | 49 ++++++++++++++++++-
 src/views/Overview/Overview.vue               | 44 +++++++++++++++++
 3 files changed, 97 insertions(+), 1 deletion(-)

diff --git a/src/locales/en-US.json b/src/locales/en-US.json
index 3acbf52..d38a22c 100644
--- a/src/locales/en-US.json
+++ b/src/locales/en-US.json
@@ -577,6 +577,11 @@
     "powerConsumption": "Power consumption",
     "serialNumber": "Serial number",
     "serverInformation": "Server information",
+    "cpuId": "CPU information",
+    "socket":"Socket",
+    "familyId":"Family ID",
+    "modelId":"Model ID",
+    "steppingId":"Stepping ID",
     "events": {
       "date": "Date",
       "description": "Description",
diff --git a/src/store/modules/ResourceManagement/PowerControlStore.js b/src/store/modules/ResourceManagement/PowerControlStore.js
index 9dbddf0..ff71239 100644
--- a/src/store/modules/ResourceManagement/PowerControlStore.js
+++ b/src/store/modules/ResourceManagement/PowerControlStore.js
@@ -1,21 +1,40 @@
 import api from '@/store/api';
 import i18n from '@/i18n';

+const UNKNOWN = 'UnKnown';
 const PowerControlStore = {
   namespaced: true,
   state: {
     powerCapValue: null,
     powerConsumptionValue: null,
+    cpuId1Value: [],
+    cpuId2Value: [],
   },
   getters: {
     powerCapValue: (state) => state.powerCapValue,
     powerConsumptionValue: (state) => state.powerConsumptionValue,
+    cpuId1Value: (state) => state.cpuId1Value,
+    cpuId2Value: (state) => state.cpuId2Value,
   },
   mutations: {
     setPowerCapValue: (state, powerCapValue) =>
       (state.powerCapValue = powerCapValue),
     setPowerConsumptionValue: (state, powerConsumptionValue) =>
       (state.powerConsumptionValue = powerConsumptionValue),
+    setcpuId1Value: (state, cpuId1) => {
+      const cpuId1Value = {};
+      cpuId1Value.familyId = cpuId1.familyId;
+      cpuId1Value.modelId = cpuId1.modelId;
+      cpuId1Value.stepId = cpuId1.stepId;
+      state.cpuId1Value = [cpuId1Value];
+    },
+    setcpuId2Value: (state, cpuId2) => {
+      const cpuId2Value = {};
+      cpuId2Value.familyId = cpuId2.familyId;
+      cpuId2Value.modelId = cpuId2.modelId;
+      cpuId2Value.stepId = cpuId2.stepId;
+      state.cpuId2Value = [cpuId2Value];
+    },
   },
   actions: {
     setPowerCapUpdatedValue({ commit }, value) {
@@ -29,9 +48,37 @@ const PowerControlStore = {
           const powerCap = powerControl[0].PowerLimit.LimitInWatts;
           // If system is powered off, power consumption does not exist in the PowerControl
           const powerConsumption = powerControl[0].PowerConsumedWatts || null;
-
+          // process CPU1 values
+          //const cpu1 = powerControl[0].CpuIDs.CPU1;
+          //const tempArr = 'family1#model1#step1';
+          const cpuId1 = [];
+          if (powerControl[0].CpuIDs.CPU1 === '') {
+            cpuId1['familyId'] = UNKNOWN;
+            cpuId1['modelId'] = UNKNOWN;
+            cpuId1['stepId'] = UNKNOWN;
+          } else {
+            var cpu1Arr = powerControl[0].CpuIDs.CPU1.split('#');
+            cpuId1['familyId'] = cpu1Arr.length > 0 ? cpu1Arr.at(0) : UNKNOWN;
+            cpuId1['modelId'] = cpu1Arr.length > 1 ? cpu1Arr.at(1) : UNKNOWN;
+            cpuId1['stepId'] = cpu1Arr.length > 2 ? cpu1Arr.at(2) : UNKNOWN;
+          }
+          // process CPU2 values
+          //const cpuId2 = powerControl[0].CpuIDs.CPU2;
+          const cpuId2 = [];
+          if (powerControl[0].CpuIDs.CPU2 === '') {
+            cpuId2['familyId'] = UNKNOWN;
+            cpuId2['modelId'] = UNKNOWN;
+            cpuId2['stepId'] = UNKNOWN;
+          } else {
+            var cpu2Arr = powerControl[0].CpuIDs.CPU2.split('#');
+            cpuId2['familyId'] = cpu2Arr.length > 0 ? cpu2Arr.at(0) : UNKNOWN;
+            cpuId2['modelId'] = cpu2Arr.length > 0 ? cpu2Arr.at(1) : UNKNOWN;
+            cpuId2['stepId'] = cpu2Arr.length > 0 ? cpu2Arr.at(2) : UNKNOWN;
+          }
           commit('setPowerCapValue', powerCap);
           commit('setPowerConsumptionValue', powerConsumption);
+          commit('setcpuId1Value', cpuId1);
+          commit('setcpuId2Value', cpuId2);
         })
         .catch((error) => {
           console.log('Power control', error);
diff --git a/src/views/Overview/Overview.vue b/src/views/Overview/Overview.vue
index 2436e39..2ac37c3 100644
--- a/src/views/Overview/Overview.vue
+++ b/src/views/Overview/Overview.vue
@@ -25,6 +25,24 @@
             </page-section>
           </b-col>
         </b-row>
+        <b-row>
+          <b-col>
+            <page-section :section-title="$t('pageOverview.cpuId')">
+              <dl>
+                <dt>{{ $t('pageOverview.socket') + ' ' + '0' }}</dt>
+                <div>{{ $t('pageOverview.familyId') + ' : ' + familyId1 }}</div>
+                <div>{{ $t('pageOverview.modelId') + ' : ' + modelId1 }}</div>
+                <div>{{ $t('pageOverview.steppingId') + ' : ' + stepId1 }}</div>
+              </dl>
+              <dl>
+                <dt>{{ $t('pageOverview.socket') + ' ' + '1' }}</dt>
+                <div>{{ $t('pageOverview.familyId') + ' : ' + familyId2 }}</div>
+                <div>{{ $t('pageOverview.modelId') + ' : ' + modelId2 }}</div>
+                <div>{{ $t('pageOverview.steppingId') + ' : ' + stepId2 }}</div>
+              </dl>
+            </page-section>
+          </b-col>
+        </b-row>
       </b-col>
       <b-col>
         <page-section :section-title="$t('pageOverview.serverInformation')">
@@ -110,6 +128,8 @@ export default {
       powerCapValue: (state) => state.powerControl.powerCapValue,
       powerConsumptionValue: (state) =>
         state.powerControl.powerConsumptionValue,
+      cpuId1Value: (state) => state.powerControl.cpuId1Value[0],
+      cpuId2Value: (state) => state.powerControl.cpuId2Value[0],
       serverManufacturer() {
         if (this.server) return this.server.manufacturer || '--';
         return '--';
@@ -122,6 +142,30 @@ export default {
         if (this.server) return this.server.serialNumber || '--';
         return '--';
       },
+      familyId1() {
+        if (this.cpuId1Value) return this.cpuId1Value.familyId || '--';
+        return '--';
+      },
+      modelId1() {
+        if (this.cpuId1Value) return this.cpuId1Value.modelId || '--';
+        return '--';
+      },
+      stepId1() {
+        if (this.cpuId1Value) return this.cpuId1Value.stepId || '--';
+        return '--';
+      },
+      familyId2() {
+        if (this.cpuId2Value) return this.cpuId2Value.familyId || '--';
+        return '--';
+      },
+      modelId2() {
+        if (this.cpuId2Value) return this.cpuId2Value.modelId || '--';
+        return '--';
+      },
+      stepId2() {
+        if (this.cpuId2Value) return this.cpuId2Value.stepId || '--';
+        return '--';
+      },
     }),
     activeHostFirmware() {
       return this.$store.getters[`firmware/activeHostFirmware`];
--
2.25.1

